Java/ds/CheckBST.java

```diff
@@ -0,0 +1,227 @@
+package neoAlgo;
+
+/*This is the solution to create a Binary Tree from user inputs taken level wise and to check if that binary tree is a Binary Search tree.This solution works for duplicate input values of nodes as well.
+
+Constraints:
+0<= NodeValue <= 10^4
+  
+Incorrect Approach:
+It is not enough to check if the left node of current node is smaller than it and the right node of current node is larger than it respectively.
+This approach would't work for the following example which is actually not a Binary Search Tree --
+     15
+    /   \
+  12     28
+ /  \
+9    24 
+
+Correct Approach:
+1. For each node we have to check, if maximum value in its left sub tree is smaller than the respective node value and the minimum value in its right sub tree is larger than the respective node value. 
```

Comment:
Reduce the length of the line. Compact sentences will help the readers to better understand the Code